lib/rouge/lexers/verilog.rb

```diff
@@ -128,8 +126,8 @@ def self.keywords_system_task
         rule %r/[0-9]*'b?[01xz_?]+/, Num::Bin
         rule %r/[0-9]*'d[0-9_?]+/, Num::Integer
         rule %r/[0-9_]+[lu]*/i, Num::Integer
-        rule %r([~!%^&*+-=\|?:<>/@{}]), Operator
-        rule %r/[()\[\],.$\#]/, Punctuation
+        rule %r([-~!%^&*+=\|?:<>/@{}]), Operator
+        rule %r/[()\[\],.$\#;]/, Punctuation
```

Comment:
I don't understand how the visual sample didn't show an error if there wasn't an express `;`? Was it being included in the `+-=` range and tokenised as an operator?

Reply:
A test case fails without `;` in Punctuation becasue `+-=` range includes `;`.

```
  1) Failure:
Rouge::Lexers::Rouge::Lexers::Verilog#test_0001_lexes the demo with no errors [/home/pocke/ghq/github.com/rouge-ruby/rouge/spec/lexers_spec.rb:15]:
      refute { filter_by_token(tokname, text, lexer).any? }
               |               |        |     |      |
               |               |        |     |      true
               |               |        |     nil
               |               |        "/**\n * Verilog Lexer\n */\nmodule Foo(\n  input logic Clk_CI,\n  input logic Rst_RBI,\n  input logic A,\n  input logic B,\n  output logic C\n);\n  logic C_DN, C_DP;\n\n  assign C = C_DP;\n\n  always_comb begin : proc_next_state\n    C_DN = A + B;\n  end\n\n  // Clocked process\n  always_ff @(posedge Clk_CI, negedge Rst_RBI) begin\n    if(~Rst_RBI) begin\n      C_DP <= 1'b0;\n    end else begin\n      C_DP <= C_DN;\n    end\n  end\nendmodule\n"
               |               "Error"
               [[<Token Error>, ";"], [<Token Error>, ";"], [<Token Error>, ";"], [<Token Error>, ";"], [<Token Error>, ";"], [<Token Error>, ";"]]
```



So `;` was tokenized as an operator, but I think punctuation is better than an operator for `;`. Because semicolon means an end of line in Verilog.

Reply:
Thank you! Basically I wanted to confirm that the range `+-=` included `;` (I realise now I could have just tested this in `irb`!). I completely agree that `;` should be punctuation.